### PR TITLE
Feature/save products

### PR DESCRIPTION
**Pull request by SaschaRadojkovic**
in this feature i add the possibility to save products in local storage and to view details again [US](https://github.com/SaschaRadojkovic/capstone-project/issues/15)

### CODE REVIEW

File: pages/product/[code].js [hunk 1]
```diff
@@ -69,24 +72,65 @@ const StyledIngredientsCard = styled.div`
...
 const StyledAllergensH3 = styled.p`
+  padding: 0.5rem;
+
+  text-align: center;
+  font-weight: bold;
+`;
+
+const StyledParagraph = styled.p`
+  display: flex;
+  flex: 0 0 50%;
```
Comment (F-Kirchhoff) [inline pages/product/[code].js:106]: here too! :D

File: pages/product/[code].js [hunk 2]
```diff
@@ -183,9 +234,34 @@ export default function DetailPage() {
...
+                          code: code,
+                        },
+                        ...savedProducts,
+                      ]);
+                    }
+                  }}
+                >
+                  <SVGIcon
+                    variant={
+                      savedProducts.some((product) => product.code === code)
```
Comment (F-Kirchhoff) [inline pages/product/[code].js:257]: you check for the existence of the product twice. Please refactor it into a constant above the return value and use that variable here and in line 242 instead.

File: pages/products/index.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+`;
+const StyledImage = styled(Image)`
+  width: 135px;
+  height: 135px;
+  object-fit: contain;
+`;
+const StyledProductCard = styled.section`
+  witdh: 50%;
+  //   margin-left: 3%;
+  //   width: 80%;
```
Comment (F-Kirchhoff) [inline pages/products/index.js:26]: You need to remove all deactivated code before stating a PR.

File: pages/products/index.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            <StyledProductCard key={product.code}>
+              <StyledProductName>{product.name}</StyledProductName>
+              <StyledButtonPosition>
+                <StyledDetailsButton
+                  onClick={() => {
+                    router.push(`/product/${product.code}`);
+                  }}
+                >
+                  <SVGIcon variant="details" width="26px" />
+                </StyledDetailsButton>
```
Comment (F-Kirchhoff) [inline pages/products/index.js:86]: This should be a link component. Please use the next/link component here instead with the `href` prop.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Sascha,
I added some suggestions for your code, have a look. Generally already good stuff!

File: components/Navigation/index.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ export const NavBarWrapper = styled.div`
 
 const NavbarLink = styled(Link)`
   display: flex;
-  flex: 0 0 33.33%;
+  flex: 0 0 25%;
```
[the remark this answers is not in the document]
Comment (SaschaRadojkovic) [reply components/Navigation/index.js]: done

File: pages/product/[code].js [hunk 6]
```diff
@@ -9,17 +9,19 @@ import additives from "../../additives.json";
...
-  object-fit: cover;
+  width: 200px;
+  height: 200px;
+  object-fit: contain;
 `;
 
 const StyledProductName = styled.h2`
-  padding-top: 1rem;
+  padding-top: 0.5rem;
+  padding-bottom: 0.5rem;
```
[the remark this answers is not in the document]
Comment (SaschaRadojkovic) [reply pages/product/[code].js]: done

File: pages/product/[code].js [hunk 7]
```diff
@@ -69,24 +72,65 @@ const StyledIngredientsCard = styled.div`
...
-  box-shadow: 0 0 10px 2px rgba(128, 128, 128, 0.25);
-  height: 50%;
+  box-shadow: 1px 4px 10px 1px rgb(127, 133, 136);
 `;
 const StyledInsideCard = styled.div`
   display: flex;
 `;
 const StyledPAdditives = styled.p`
   display: flex;
+  flex: 0 0 50%;
```
[the remark this answers is not in the document]
Comment (SaschaRadojkovic) [reply pages/product/[code].js]: done

File: pages/product/[code].js (hunk 1 above)
Comment (SaschaRadojkovic) [reply pages/product/[code].js]: done ;)

File: pages/product/[code].js (hunk 2 above)
Comment (SaschaRadojkovic) [reply pages/product/[code].js]: done .thanks

File: pages/products/index.js (hunk 3 above)
Comment (SaschaRadojkovic) [reply pages/products/index.js]: done

File: pages/products/index.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+  cursor: pointer;
+  color: inherit;
+`;
+
+const StyledButtonPosition = styled.div`
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const StyledInsideCard = styled.div``;
```
[the remark this answers is not in the document]
Comment (SaschaRadojkovic) [reply pages/products/index.js]: done

File: pages/products/index.js (hunk 4 above)
Comment (SaschaRadojkovic) [reply pages/products/index.js]: done

> Hey Sascha, I added some suggestions for your code, have a look. Generally already good stuff!

thaanks :)

File: pages/product/[code].js [hunk 9]
```diff
@@ -183,9 +236,32 @@ export default function DetailPage() {
         <>
           <StyledAllCards>
             <StyledProductCard>
-              <StyledProductName>
-                {data.product.product_name}{" "}
-              </StyledProductName>
+              <StyledProductName>{data.product.product_name}</StyledProductName>
+              <StyledButtonDiv>
+                <StyledSaveButton
```
Comment (NikRoe) [inline pages/product/[code].js:241]: please be explicit about the type of button
```
type="button"
```

**Review by NikRoe — APPROVED** [review]
Hi there,

looking good 💪

File: pages/product/[code].js (hunk 9 above)
Comment (SaschaRadojkovic) [reply pages/product/[code].js:241]: ok i will !! :) thanks

**Pull request merged**